decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
```

Comment:
-[ ] **REQUIRED:** The correct name method should return the result of the correct_name method of the @namable is implemented